### PR TITLE
feat(config): support loading intl messages that use TX Structured JSON format

### PR DESCRIPTION
**Pull request by emmenko**
As we're looking into adopting [Transifex's Structured JSON format](https://help.transifex.com/en/articles/6220899-structured-json), we can provide better support for it in our tools.

### CODE REVIEW

**Review by dogayuksel — APPROVED** [review]
Hey, thank you for looking into this. Menu substitution should work now. Only concern is Transifex's structured_json implementation supports nesting, though not specified how deep. From [their docs](https://help.transifex.com/en/articles/6220899-structured-json) slightly modified:
```
...
   "key_1":{
      "key_nested_1a":{
         "string":"localize"
      },
      "key_nested_1b":{
         "key_nested_1ba":{
            "string":"translation tools"
         },
         "key_nested_1bb":{
            "string":"some message"
         }
...
```

I don't think it is necessary for us to adopt this, even their example looks a bit confusing. However we should mention it on the docs that our `structured_json` approach expects a flat list, so that custom application customers know what to expect.

File: packages/application-config/src/substitute-variable-placeholders.ts [hunk 1]
```diff
@@ -31,6 +40,11 @@ const isIntlVariablePlaceholder = (valueOfPlaceholder: string) =>
 const isFilePathVariablePlaceholder = (valueOfPlaceholder: string) =>
   Boolean(valueOfPlaceholder.match(filePathRefSyntax));
 
+const isStructuredJson = (
```
Comment (CarlosCortizasCT) [inline packages/application-config/src/substitute-variable-placeholders.ts:43]: Could we maybe use the new exported function from the `i18n` package instead?
(_maybe we don't want to add that dependency to this package_)

**Review by CarlosCortizasCT — APPROVED** [review]
Cool 😎

**Review by tdeekens — APPROVED** [review]
Thanks. This should do the trick.

File: packages/application-config/src/substitute-variable-placeholders.ts (hunk 1 above)
Comment (emmenko) [reply packages/application-config/src/substitute-variable-placeholders.ts:43]: Yeah I didn't want to include an extra dependency. I think it's fine to have it duplicated for now.

**Comment by emmenko** [comment]
> However we should mention it on the docs that our structured_json approach expects a flat list, so that custom application customers know what to expect.

Thanks @dogayuksel , good point.

We'll update the docs as a follow up, first I wanted us to adopt it internally and make sure it works properly.

**Pull request merged**